### PR TITLE
SCSS variable structure - integrate CSS custom properties part 2

### LINKED ISSUE #70: Combine mixins for global use
## Description
We're currently using mixins within the Sass modules where they're being consumed. Whenever possible, we should consolidate applicable mixins into the global `_mixins.scss` file for easier reference and reuse.

### PR DESCRIPTION
**Pull request by teenwolfblitzer**
## Description
PART DEUX

- Refactor and streamline use of Sass variables in stylesheets.
- Relocate mixins to global stylesheet
- Documentation now has a separate set of variables - a future issue may expand on this

Eventually, we may implement additional themes for Sage, including dark mode. Replacing/augmenting with CSS custom props as necessary would be extremely helpful, though this is likely to be dependent on future IE11 support.

## Related Issues
- Resolves #70 
- Closes remainder of #46

### CODE REVIEW

File: app/assets/stylesheets/sage/patterns/objects/_tabs.scss [hunk 1]
```diff
@@ -6,36 +6,43 @@
...
+    transition: color $sage-tabs-default-transition;
+
+    &:last-of-type {
+      margin-right: 0;
+    }
+
+    &::after {
+      content: "";
+      display: inline-block;
+      // transform: scaleY(0);
```
Comment (philschanely) [inline app/assets/stylesheets/sage/patterns/objects/_tabs.scss]: Are these needed here?

File: app/assets/stylesheets/sage_docs/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// :root {
```
Comment (philschanely) [inline app/assets/stylesheets/sage_docs/_variables.scss:11]: Add a comment above for why this is here?

**Review by philschanely — APPROVED** [review]
Two small suggestions. 👍

File: app/assets/stylesheets/sage/core/_fonts.scss [hunk 3]
```diff
@@ -1,14 +1,7 @@
...
+/* ==================================================
+  ** _fonts.scss
 
-// ==================================================
-// Sources
-// ==================================================
+================================================== */
 
 @font-face {
   font-family: "Inter";
```
Comment (AndrwM) [inline app/assets/stylesheets/sage/core/_fonts.scss:7]: Luv this typeface.

File: app/assets/stylesheets/sage/core/_variables.scss [hunk 4]
```diff
@@ -119,3 +144,66 @@ $sage-switch-focus-outline-width: 1 !default;
...
+
+
+// ==================================================
+// ASSISTANT
+// ==================================================
+
+$sage-assistant-height: 3.5rem !default;
+$sage-assistant-branding-height: 1.25rem !default; // 20px
+$sage-assistant-search-height: 2.5rem !default; // 40px
+$sage-assistant-search-bg-color: sage-color(grey, 300) !default;
```
Comment (AndrwM) [inline app/assets/stylesheets/sage/core/_variables.scss:197]: Really excited about these color selection functions

File: app/assets/stylesheets/sage/patterns/objects/_tabs.scss [hunk 5]
```diff
@@ -6,36 +6,43 @@
...
-      }
+      padding-bottom: sage-spacing(xs);
+      color: $sage-tabs-active-color;
+      font-weight: $sage-tabs-active-font-weight;
     }
 
-    &:last-of-type {
-      margin-right: 0;
+    &--active::after {
+      // transform: none;
```
Comment (AndrwM) [inline app/assets/stylesheets/sage/patterns/objects/_tabs.scss]: Remove commented code

**Review by AndrwM — APPROVED** [review]
This organization is a dream.

File: app/assets/stylesheets/sage/patterns/objects/_tabs.scss (hunk 1 above)
Comment (teenwolfblitzer) [reply app/assets/stylesheets/sage/patterns/objects/_tabs.scss]: Whoops, left those in by mistake

**Pull request merged**